### ISSUE TITLE
BigQueryType should skip None fields instead of converting them to null

### ISSUE THREAD

**Issue by flylo** [spotify/scio#388]
If we use `JobTest` to test a simple pipeline that reads from a nullable BigQuery field into a `case class` with `Option[T]` fields, `BigQueryType.toTableRow[T]` throws a `java.lang.NullPointerException`. The standard (non-testing) interface for `BigQueryType.toTableRow[T]` converts nulls to Options, so this behavior seems like a bug.

When the example `JobTest` test is run for the following job, we get a NPE.
```scala
import com.spotify.scio.{ContextAndArgs, ScioContext}
import com.spotify.scio.experimental._
import com.spotify.scio.values.SCollection

object ReproducingIssue {

  @BigQueryType.toTable
  case class MyNullableCaseClass(nullableField: Option[String])

  def querySomeTable(sc: ScioContext, tableRef: String): SCollection[MyNullableCaseClass] = {
    sc.bigQueryTable(tableRef).map(BigQueryType.fromTableRow[MyNullableCaseClass])
  }

  def main(cmdlineArgs: Array[String]): Unit = {
    val (sc, args) = ContextAndArgs(cmdlineArgs)

    querySomeTable(sc, "project:dataset.some_table")
      .saveAsTypedBigQuery("project:dataset.some_other_table")

    sc.close()
  }

}
```

The NPE is thrown when we call `.map(BigQueryType.fromTableRow[MyNullableCaseClass])`.

The test that we run to generate this error is as follows:
```scala
import com.spotify.cream.scio.lookalike.pipelines.ReproducingIssue.MyNullableCaseClass
import com.spotify.scio.testing.{BigQueryIO, JobTest, PipelineSpec}
import org.scalatest.FlatSpec
import com.spotify.scio.experimental._

class ReproducingIssueTest extends FlatSpec with PipelineSpec {

  "This" should "reproduce issue" in {

    val mockTableInput = List(MyNullableCaseClass(Option.empty[String]))
      .map(BigQueryType.toTableRow[MyNullableCaseClass])

    JobTest[ReproducingIssue.type]
      .input(BigQueryIO("project:dataset.some_table"), mockTableInput)
      .output(BigQueryIO("project:dataset.some_other_table")) {
        actual =>
          actual should not be None
      }
      .run()
  }

}
```

**Comment by nevillelyh** [comment]
Seems DF doesn't like `null` in a BigQuery `TableRow` field. We might have to change `toTableRow` behavior to ignore `null` fields instead of encoding `key -> null`.

Stacktrace:
```
[info]   com.google.cloud.dataflow.sdk.Pipeline$PipelineExecutionException: java.lang.NullPointerException
[info]   at com.google.cloud.dataflow.sdk.runners.inprocess.InProcessPipelineRunner.run(InProcessPipelineRunner.java:279)
[info]   at com.google.cloud.dataflow.sdk.runners.inprocess.InProcessPipelineRunner.run(InProcessPipelineRunner.java:69)
[info]   at com.google.cloud.dataflow.sdk.Pipeline.run(Pipeline.java:181)
[info]   at com.google.cloud.dataflow.sdk.testing.TestPipeline.run(TestPipeline.java:113)
[info]   at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:320)
[info]   at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:312)
[info]   at com.spotify.scio.ScioContext.requireNotClosed(ScioContext.scala:360)
[info]   at com.spotify.scio.ScioContext.close(ScioContext.scala:312)
[info]   at com.spotify.scio.examples.BQTest$.main(BQTest.scala:41)
[info]   at com.spotify.scio.examples.BQTest.main(BQTest.scala)
[info]   ...
[info]   Cause: java.lang.NullPointerException:
[info]   at com.google.api.client.util.ArrayMap$Entry.hashCode(ArrayMap.java:419)
[info]   at java.util.AbstractMap.hashCode(AbstractMap.java:530)
[info]   at java.util.Arrays.hashCode(Arrays.java:4146)
[info]   at java.util.Objects.hash(Objects.java:128)
[info]   at com.google.cloud.dataflow.sdk.util.WindowedValue$ValueInGlobalWindow.hashCode(WindowedValue.java:254)
[info]   at java.util.HashMap.hash(HashMap.java:338)
[info]   at java.util.HashMap.get(HashMap.java:556)
[info]   at com.google.cloud.dataflow.sdk.repackaged.com.google.common.collect.AbstractMapBasedMultimap.put(AbstractMapBasedMultimap.java:193)
[info]   at com.google.cloud.dataflow.sdk.repackaged.com.google.common.collect.AbstractSetMultimap.put(AbstractSetMultimap.java:128)
[info]   at com.google.cloud.dataflow.sdk.repackaged.com.google.common.collect.HashMultimap.put(HashMultimap.java:49)
```

**Comment by nevillelyh** [comment]
Fixed in #391
